src/overlay_94.c

```diff
@@ -0,0 +1,204 @@
+#include "assert.h"
+#include "heap.h"
+#include "party.h"
+#include "pokemon.h"
+#include "constants/pokemon.h"
+#include "constants/species.h"
+#include "constants/sndseq.h"
+#include "overlay_94.h"
+
+void ov94_021E5900(struct UnkStruct_Overlay_94_A* unkPtr) {
+    if ((u32) unkPtr->unkc80 != 0) {
+        GF_ASSERT(FALSE);
+    }
+    unkPtr->unkc80 = AllocFromHeap(0xc, 0x1c);
+    MI_CpuFill8(unkPtr->unkc80, 0, 0x1c);
```

Comment:
```suggestion
    MI_CpuClear8(unkPtr->unkc80, sizeof(struct UnkStruct_Overlay_94_C));
``` (also double-check the types, the size of 0x1C does not match the size of the struct)

Reply:
Ahh yup, I accidentally merged two structs together since the later only used variables stored past 1c. Nice catch!